src/composables/PluginManager.js

```diff
@@ -0,0 +1,108 @@
+import { readTextFile } from 'src/composables/Files';
+import plugins from 'src/plugins';
+
+let instanciatePlugins = [];
+
+export function initPluginMetadata(plugin) {
+  plugin.metadata = new plugin.pluginModel.PluginMetadata();
+
+  const definitions = plugin.metadata.getDefinitions();
+
+  // Add url file in svgImage for componentDefinitionCard
+  definitions.components.forEach((def) => {
+    if (def.icon) {
+      def.svgImage = `/plugins/${plugin.name}/icons/${def.icon}.svg`;
+    }
+  });
+  definitions.links.forEach((def) => {
+    if (def.icon) {
+      def.svgImage = `/plugins/${plugin.name}/icons/${def.icon}.svg`;
+    }
+  });
+
+  plugin.definitions = definitions;
+}
+
+export function getFilesInfo(plugin, defType) {
+  const definitions = plugin.definitions[defType].filter((c) => c.blockType !== 'provider');
+  return definitions.reduce((acc, def) => {
+    if (def.model) {
+      acc.push({
+        name: def.model,
+        type: 'models',
+        path: `/plugins/${plugin.name}/models/${def.model}.svg`,
+      });
+    }
+
+    if (def.icon) {
+      acc.push({
+        name: def.icon,
+        type: 'icons',
+        path: `/plugins/${plugin.name}/icons/${def.icon}.svg`,
+      });
+    }
+
+    return acc;
+  }, []);
+}
+
+// Remove if svg import is possible
+export async function createPluginResources(plugin) {
+  const files = [...new Set(
+    getFilesInfo(plugin, 'components')
+      .concat(getFilesInfo(plugin, 'links')),
```

Comment:
no need to ge files from link because we don't have icon or model on link

Reply:
Done